src/go/rpk/pkg/cli/cmd/wasm/deploy_test.go

```diff
@@ -0,0 +1,118 @@
+package wasm
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/Shopify/sarama"
+	"github.com/Shopify/sarama/mocks"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/require"
+)
+
+type FileInformationTest struct {
+	name	string
+	content	string
+}
+
+func TestNewDeployCommand(t *testing.T) {
+	tests := []struct {
+		name		string
+		producer	func(bool, int32) (sarama.SyncProducer, error)
+		fileInformation	FileInformationTest
+		args		[]string
+		expectedOutput	[]string
+		expectedErr	string
+		pre		func(t *testing.T, fs afero.Fs, fileInformation FileInformationTest)
+		failSendMessage	bool
+	}{
+		{
+			name:	"it should publish a message with correct format",
+			args:	[]string{"--description", "coprocessor description"},
+			fileInformation: FileInformationTest{
+				name:		"fileName.js",
+				content:	"let s = 'text'",
+			},
+			pre: func(t *testing.T, fs afero.Fs, fileInformation FileInformationTest) {
+				require.NoError(t, createMockFile(fs, t, fileInformation.name, fileInformation.content))
+			},
+		}, {
+			name:	"it should fail if the file extension isn't js",
+			fileInformation: FileInformationTest{
+				name: "fileName.html",
+			},
+			expectedErr:	"can't deploy 'fileName.html': only .js files are supported.",
+		}, {
+			name:	"it should fail if the file doesn't exist",
+			fileInformation: FileInformationTest{
+				name: "fileName.js",
+			},
+			expectedErr:	"open fileName.js: file does not exist",
+		}, {
+			name:	"it should show an error if there is a error on send message",
+			fileInformation: FileInformationTest{
+				name: "fileName.js",
+			},
+			pre: func(t *testing.T, fs afero.Fs, fileInformation FileInformationTest) {
+				require.NoError(t, createMockFile(fs, t, fileInformation.name, fileInformation.content))
+			},
+			expectedErr: "Error on deploy fileName.js, message error: kafka mock error," +
+				" please check your redpanda instance ",
+			failSendMessage:	true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			createProduce := func(jvm bool, partition int32) (sarama.SyncProducer, error) {
+				produce := mocks.NewSyncProducer(t, nil)
+				if tt.failSendMessage == false {
+					produce.ExpectSendMessageWithCheckerFunctionAndSucceed(func(val []byte) error {
+						if tt.fileInformation.content != "" {
+							require.True(t, bytes.Contains(val, []byte(tt.fileInformation.content)))
```

Comment:
Use `require.Equal(t, string(val), tt.fileInfo.content)`

Reply:
I need to use `bytes.Contains`, because `val` has more information

Reply:
with the new format message, I changed it and I use `require.Equal`